OpenSim/Common/DataTable.h

```diff
@@ -0,0 +1,408 @@
+/* -------------------------------------------------------------------------- *
+ *                            OpenSim:  DataTable.h                           *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2015 Stanford University and the Authors                *
+ * Authors:                                                                   *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#ifndef OPENSIM_DATA_TABLE_H_
+#define OPENSIM_DATA_TABLE_H_
+
+/** \file
+This file defines the  DataTable_ class, which is used by OpenSim to provide an 
+in-memory container for data access and manipulation.                         */
+
+// Non-standard headers.
+#include "SimTKcommon.h"
+#include "OpenSim/Common/Exception.h"
+#include "OpenSim/Common/ValueArrayDictionary.h"
+
+namespace OpenSim {
+
+class InvalidRow : public Exception {};
+
+class IncorrectNumColumns : public InvalidRow {
+public:
+    IncorrectNumColumns(const std::string& file,
+                        size_t line,
+                        const std::string& func,
+                        size_t expected,
+                        size_t received) {
+        std::string msg{errorMessagePrefix(file, line, func)};
```

Comment:
It didn't want to edit the OpenSim::Exception class in this PR. Because it is used everywhere in OpenSim, changes may become invasive to other code.
I think we should make required changes in a separate PR for error handling.
@aseth1 @chrisdembia -- what do you think ?